### PR TITLE
PODS-1368 : PSA Reg - NON UK - Partnership - NAVIGATION

### PR DESCRIPTION
**Pull request by abhijeetardale**
PODS-1368 : PSA Reg - NON UK - Partnership - NAVIGATION

### CODE REVIEW

**Comment by abhijeetardale** [comment]
we got wrong branch mistakenly, so renamed with correct ticket number

File: app/utils/navigators/PartnershipNavigator.scala [hunk 1]
```diff
@@ -94,26 +103,58 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
   }
 
   private def addressYearsIdRoutes(answers: UserAnswers): Option[NavigateTo] = {
```
[the remark this answers is not in the document]
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: fixed

File: app/utils/navigators/PartnershipNavigator.scala [hunk 2]
```diff
@@ -94,26 +103,58 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+    (answers.get(PartnershipAddressYearsId), answers.get(AreYouInUKId)) match {
+      case (Some(AddressYears.UnderAYear), Some(true)) => NavigateTo.save(routes.PartnershipPreviousAddressPostCodeLookupController.onPageLoad(NormalMode))
+      case (Some(AddressYears.UnderAYear), None) => NavigateTo.save(routes.PartnershipPreviousAddressPostCodeLookupController.onPageLoad(NormalMode))
+      case (Some(AddressYears.UnderAYear), Some(false)) => NavigateTo.save(routes.PartnershipPreviousAddressController.onPageLoad(NormalMode))
+      case (Some(AddressYears.OverAYear), _) => NavigateTo.save(routes.PartnershipContactDetailsController.onPageLoad(NormalMode))
+      case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
 
   private def addressYearsCheckIdRoutes(answers: UserAnswers): Option[NavigateTo] = {
```
[the remark this answers is not in the document]
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: true

File: app/utils/navigators/PartnershipNavigator.scala [hunk 3]
```diff
@@ -94,26 +103,58 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+    (answers.get(PartnershipAddressYearsId), answers.get(AreYouInUKId)) match {
+      case (Some(AddressYears.UnderAYear), Some(true)) => NavigateTo.save(routes.PartnershipPreviousAddressPostCodeLookupController.onPageLoad(CheckMode))
+      case (Some(AddressYears.UnderAYear), None) => NavigateTo.save(routes.PartnershipPreviousAddressPostCodeLookupController.onPageLoad(CheckMode))
+      case (Some(AddressYears.UnderAYear), Some(false)) => NavigateTo.save(routes.PartnershipPreviousAddressController.onPageLoad(CheckMode))
+      case (Some(AddressYears.OverAYear), _) => NavigateTo.save(routes.CheckYourAnswersController.onPageLoad())
+      case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
 
   private def sameContactAddress(mode: Mode, answers: UserAnswers): Option[NavigateTo] = {
```
[the remark this answers is not in the document]
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: true

File: app/utils/navigators/PartnershipNavigator.scala [hunk 4]
```diff
@@ -94,26 +103,58 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+      case (Some(false), Some(false)) => NavigateTo.save (routes.PartnershipContactAddressController.onPageLoad (mode) )
+      case (Some(false), None) => NavigateTo.save(routes.PartnershipContactAddressPostCodeLookupController.onPageLoad(mode))
       case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
-}
\ No newline at end of file
+
+
+  private def regionBasedNameNavigation(answers: UserAnswers): Option[NavigateTo] = {
```
[the remark this answers is not in the document]
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: its independent of feature toggle, hence not updated

File: app/utils/navigators/PartnershipNavigator.scala [hunk 5]
```diff
@@ -94,26 +103,58 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+      countryOptions.regions(address.country.getOrElse("")) match {
+        case UK => NavigateTo.dontSave(controllers.register.routes.BusinessTypeAreYouInUKController.onPageLoad(CheckMode))
+        case EuEea => NavigateTo.dontSave(routes.WhatYouWillNeedController.onPageLoad())
+        case RestOfTheWorld => NavigateTo.dontSave(routes.OutsideEuEeaController.onPageLoad())
+        case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
+      }
+    }
+  }
+
+  private def regionBasedContactDetailsRoutes(answers: UserAnswers): Option[NavigateTo] = {
```
[the remark this answers is not in the document]
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: fixed

**Comment by PaulCDurham** [comment]
You have eight commits since I approved this. You need to get this PR re-approved

File: test/identifiers/register/AreYouInUKIdSpec.scala [hunk 6]
```diff
@@ -21,6 +21,7 @@ import java.time.LocalDate
 import identifiers.register.company.{BusinessDetailsId, CompanyAddressId}
 import identifiers.register.individual.{IndividualAddressId, IndividualDateOfBirthId, IndividualDetailsCorrectId, IndividualDetailsId}
 import identifiers.register.partnership.{PartnershipDetailsId, PartnershipRegisteredAddressId}
+import identifiers.register.partnership.{PartnershipDetailsId, PartnershipRegisteredAddressId}
```
Comment (PaulCDurham) [inline test/identifiers/register/AreYouInUKIdSpec.scala]: Remove duplicate

Comment (abhijeetardale) [reply test/identifiers/register/AreYouInUKIdSpec.scala]: fixed

**Pull request merged**